app/models/main.py

```diff
@@ -60,6 +60,35 @@ def __init__(self, astext_type=None):
         super(JSON, self).__init__(none_as_null=True, astext_type=astext_type)
 
 
+class RemovePersonalDataModel:
+    """This should be added to classes we wish to remove personal data from."""
+    personal_data_removed = db.Column(db.Boolean, index=False, unique=False, nullable=False, default=False)
+
+    def remove_personal_data(self):
+        """Implement this method removing personal data from the inheriting object.
+
+        It should at some point set the 'personal_data_removed' flag to True
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    def validate_personal_data_removed(mapper, connection, instance):
+        # """The only time we should be able to update an object with """
+        session = object_session(instance)
+        model_class = instance.__class__
+        if session.query(model_class.personal_data_removed).filter(model_class.id == instance.id).scalar():
+            raise ValidationError("Cannot update an object once personal data has been removed")
+
+
+listen(
```

Comment:
Had to manually write an event listener instead of using the `@validates` decorator because we need this to be triggered on every save of an object.

Reply:
`@validates` decorators only trigger when their field is changed